src/execution_position.rs

```diff
@@ -0,0 +1,79 @@
+use yultsur::yul::{FunctionCall, Identifier, IdentifierID, SourceLocation};
+
+use crate::smt::{self, SMTExpr};
+use crate::ssa_tracker::SSATracker;
+use crate::variables::SystemVariable;
+
+/**
+ * Stores the current call stack encoded in a single number that can be used
+ * during the SMT encoding.
+ *
+ * Can be extended in the future with other information like loop iterations.
+ */
+#[derive(Default)]
+pub struct ExecutionPositionManager {
```

Comment:
Why not call this whole thing `callstack` or similar?

Reply:
Because I wanted to add other things to this like iteration number in a loop unroll. (see the documentation)